### DISCUSSION TITLE
Cannot find a service for the path

### DISCUSSION THREAD

**Discussion by LiamWalsh98** [eclipse-glsp/glsp#558] [Q&A]
This seems like it should be relatively straightforward but I can't figure out why my plugin is no longer being run. 

`root ERROR Cannot find a service for the path: /services/glsp/RTPoet/:id`

I believe this began after I began extending the GLSPDiagramLanguage to provide the id, label, extensions, etc. The server plugin is unchanged, and the attributes of the extended GLSPDiagramLanguage are being used in all the same places. Any help would be appreciated.

**Comment by tortmayr** [comment]
HI @LiamWalsh98  sorry for the late response.
Without any additional information it is kind of hard to exactly track down  the problem.
Is the mentioned error logged in the front-end (i.e. browser) or in the back-end process of Theia?

**Reply by LiamWalsh98** [reply]
Hi! Not a problem. It's true that this is a bit vague, sorry about that.

The error is coming from theia's logs. After some digging I came to realize the message is coming from a @theia/core's [MessagingContribution:145](https://github.com/eclipse-theia/theia/blob/41a2f5b8d330944498a8e7715d8a30921a73f5d6/packages/core/src/node/messaging/messaging-contribution.ts#L145). Unfortunately the stacktrace is not much use because it seems to begin by receiving a message from a source unknown to me.

So it does seem to be in the scope of Theia rather than GLSP. Theia seems to start just fine, but when a client connects via browser it logs the aforementioned error. 

Alongside the message that "can't find a service" there are several that are sent in the same way whenever a new client connects. (all sequential values of `message` in the context of the above linked LOC)

```
{"kind":"open","id":0,"path":"/services/commands"}
{"kind":"open","id":1,"path":"/services/logger"}
{"kind":"open","id":2,"path":"/services/messageService"}
{"kind":"open","id":3,"path":"/services/remote-filesystem"}
{"kind":"open","id":4,"path":"/services/envs"}
{"kind":"open","id":5,"path":"/services/workspace"}
{"kind":"open","id":6,"path":"/services/application"}
{"kind":"open","id":7,"path":"/services/i18n"}
{"kind":"open","id":8,"path":"/services/shell-terminal"}
{"kind":"open","id":9,"path":"/services/glsp/RTPoet/:id"}
```

RTPoet is the [`contributionId`](https://github.com/eclipse-glsp/glsp-theia-integration/blob/82a6df6cd9d984e107b0e6f386c231dfa412f92c/packages/theia-integration/src/common/glsp-diagram-language.ts#L23) of my language, for reference.

It really just seems like it's not being loaded properly. But I don't think it has anything to do with nomenclature, because my frontend and backend modules both make use of the GLSPDiagramLanguage attributes. It's possible I have something misconfigured somewhere else though.

**Reply by tortmayr** [reply]
That does look strange to me. Are you sure that you have properly configured the `GLSPTheiaFrontendModule` for your GLSP language?
(see also https://github.com/eclipse-glsp/glsp-theia-integration/blob/3655cc2dcb7836fcb191e96784bcea645b56664e/examples/workflow-theia/src/browser/workflow-frontend-module.ts#L30).

**Reply by LiamWalsh98** [reply]
I have overridden `bindGLSPClientContribution`, `bindDiagramConfiguration`, `bindTheiaGLSPConnector`, and `configureDiagramManager`. The first two are identical to the workflow example (and the third with respect to the other two, but obviously that's not in the workflow example). The last one is implemented the same as the documentation example found in [`GLSPTheiaFrontEndModule.configureDiagramManager`](https://github.com/eclipse-glsp/glsp-theia-integration/blob/6e2eb4309a543620066e08aa4a1458071f63f010/packages/theia-integration/src/browser/glsp-theia-container-module.ts#L133-L134). It seems to me like it's right, but I can't be sure. 

I had realized yesterday that perhaps there's a better way to debug theia plugins rather than just using `theia inspect`? That method doesn't seem to pick up on the frontend module's code (even in the ecore-glsp example). I was hoping to be able to figure out if there's an entry point to the frontend code that precedes the `GLSPTheiaFrontendModule`. Maybe that could better help me diagnose the real issue.

Anyway, I very much appreciate the response as always. Thank you!
